src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -501,6 +510,20 @@ protected void addKnnDoc(String index, String docId, List<String> fieldNames, Li
         assertEquals(request.getEndpoint() + ": failed", RestStatus.CREATED, RestStatus.fromCode(response.getStatusLine().getStatusCode()));
     }
 
+    /**
+     * Adds a doc where document is represented as a string.
+     */
+    protected void addKnnDoc(final String index, final String docId, final String document) throws IOException {
+        Request request = new Request("POST", "/" + index + "/_doc/" + docId + "?refresh=true");
+
+        request.setJsonEntity(document);
+        client().performRequest(request);
+
+        request = new Request("POST", "/" + index + "/_refresh");
```

Comment:
We are already passing refresh paramater = true on line 517. Is this API still needs to be called?

Reply:
ack